src/s1reader/s1_burst_slc.py

```diff
@@ -431,3 +420,35 @@ def as_dict(self):
                 val = temp
             self_as_dict[key] = val
         return self_as_dict
+
+    @property
+    def sensing_mid(self):
+        '''Returns sensing mid as datetime.datetime object.
+
+        Returns:
+        --------
+        _ : datetime.datetime
+            Sensing mid as datetime.datetime object.
+        '''
+        d_seconds = 0.5 * (self.shape[0] - 1) * self.azimuth_time_interval
+        return self.sensing_start + datetime.timedelta(seconds=d_seconds)
+
+    @property
+    def sensing_stop(self):
+        '''Returns sensing end as datetime.datetime object.
+
+        Returns:
+        --------
+        _ : datetime.datetime
+            Sensing end as datetime.datetime object.
+        '''
+        d_seconds = (self.shape[0] - 1) * self.azimuth_time_interval
```

Comment:
Could you help understand why there is `- 1` here for both sensing_stop and sensing_mid?

Reply:
The `- 1` accounts for the first element. Without the `- 1`, the `sensing_stop` returns a time value outside radar grid.

Reply:
makes sense for stop but not for mid.

Reply:
For `sensing_stop`, I feel like this is similar to "a pixel is an area or a point" topic, if `sensing_start` is the beginning of the first pixel and `sensing_stop` is the end of the last pixel, then we should not use `- 1`, am I understanding it right?

@LiangJYu could you maybe show the value of the sensing_stop and radar grid boundaries here for us to better understand?

Reply:
![image](https://user-images.githubusercontent.com/2991893/165602342-a7931af6-a302-49d8-9fc3-a85ce721676f.png)
[compute sensing stop.xlsx](https://github.com/opera-adt/s1-reader/files/8575769/compute.sensing.stop.xlsx)

Reply:
@yunjunz sensing_stop here is actually the time-tag of the last sample. This is just the conversion between index and time. If sensing_start represents the beginning of the first pixel, sensing stop represents the beginning of the last pixel.

Reply:
Thank you @LiangJYu and @hfattahi for the explanation.